src/inventory/furnace_inventory.rs

```diff
@@ -0,0 +1,165 @@
+use crate::inventory::slot_mapping::SlotMapping;
+use crate::inventory::{Inventory, Item};
+use crate::render::hud::Hud;
+use crate::render::inventory::InventoryWindow;
+use crate::render::Renderer;
+use crate::ui;
+use crate::ui::{Container, HAttach, VAttach};
+use std::sync::Arc;
+
+use parking_lot::RwLock;
+
+pub struct FurnaceInventory {
+    slots: SlotMapping,
+    client_state_id: i16,
+    name: String,
+    id: i32,
+}
+
+impl FurnaceInventory {
+    pub fn new(
+        renderer: Arc<Renderer>,
+        base_slots: Arc<RwLock<SlotMapping>>,
+        name: String,
+        id: i32,
+    ) -> Self {
+        let mut slots = SlotMapping::new((176, 166));
+        slots.set_child(base_slots, (8, 84), (3..39).collect());
+
+        // Ingredient slot
+        slots.add_slot(0, (56, 17));
+
+        // Fuel slot
+        slots.add_slot(1, (56, 53));
+
+        // Output slot
+        slots.add_slot(2, (116, 35));
```

Comment:
maybe we could sole the output slot stuff by having a ruleset that can be connected to a slot, so that we have the problem generally solved and can enable plugins/mods to very easily add such slots themselves

Reply:
Maybe we could have smth like an enum with an variant for "normal" slots and for "special" slots and have these special slots just handle user inputs directly (but we might provide some helpers to do so) - this is just an idea, idk if you (or anybody else really) can think of something better

Reply:
Yeah, I was thinking about doing something like that. I'll see what I can get working, but I probably put those changes on a different PR.